trino/trino.go

```diff
@@ -1043,6 +1125,22 @@ func (qr *driverRows) initColumns(qresp *queryResponse) error {
 	return nil
 }
 
+func (qr *driverRows) updateCallback(qrStats QueryProgressCallbackResponse) {
+	currentTime := time.Now()
+	diff := currentTime.Sub(qr.stmt.conn.callbackPeriod.LastCallbackTime)
+	period := qr.stmt.conn.callbackPeriod.Period
+	// Check if period has passed OR if query state changes
+	if diff >= period || qr.stmt.conn.callbackPeriod.LastQueryState != qrStats.QueryStats.State {
```

Comment:
Reverse this condition and do an empty return here to avoid one level of indentation of the remaining code.